### DISCUSSION TITLE
Auto find slide image-boundaries in high-res slide scans, with skewness, and black-content in image.

### DISCUSSION THREAD

**Discussion by vincedarley** [ImageMagick/ImageMagick#2645] [Help] [answered]
I have a couple of thousand slides I scanned some time ago into 16-bit tiffs, and I'm now trying to process them more usefully.

Each slide was scanned with a black border around the image. Sometimes this isn't totally oriented correctly, so I need to de-skew them. I do know that, in every slide the image I want is basically the same size (e.g. 5300 x 3600 pixels). And some of the images (e.g. one of the two below), contains a lot of black so it is somewhat tricky to find the border.

Any suggestions as to how to robustly get a rectangle, potentially skewed (and/or a rotation angle) so that I can reliably extract just the image from each of these?

I've been experimenting a lot with 'canny' and 'hough-lines' and a variety of the nice utilities from Fred's ImageMagick Scripts, but somehow nothing quite gets what I need on any of the more complex images contains black, skewed, etc.

Here are a couple of links to compressed jpegs of the images, as an example. (Can happily provide links to the original tiffs too, but they are very big!).

Thanks for any tips or help...

![slide1](https://user-images.githubusercontent.com/25983202/94568661-b3b68300-0264-11eb-884a-d69f5388b5f1.jpeg)

https://app.box.com/s/m8ulbpotzex04l1nb68m6tnkpzko7jm4
https://app.box.com/s/9169h2pv2ea6a0iuiuteewtj77nsp1tq

**Comment by fmw42** [comment, marked as answer]
See minimum rotated bounding box at https://imagemagick.org/script/convex-hull.php. (only available in IM 7)

 `magick img.jpeg -fuzz 25% -set option:angle "%[minimum-bounding-box:unrotate]" -background black -rotate "%[angle]" -trim x.jpg`

![x](https://user-images.githubusercontent.com/13458218/94589599-4ee63300-023a-11eb-9766-0b54fa1b97f5.jpg)

**Comment by vincedarley** [comment]
Thank you - that's great. With a bit of experimentation on the fuzz-factor for some of the images, this is working extremely well.

**Reply by fmw42** [reply]
Glad to hear that. The fuzz factor is important as you have found out. It is gratifying to know that some one is using this new tool. I thought it might be useful for just this purpose.
